typedargs/metadata.py

```diff
@@ -42,11 +45,8 @@ def __init__(self, func, name=None):
         # If we are called to annotate a context, we won't necessarily
         # have any arguments
         try:
-            if sys.version_info.major >= 3:
-                spec = inspect.getfullargspec(func)
-                args, varargs, kwargs, defaults = spec[:4]
-            else:
-                args, varargs, kwargs, defaults = inspect.getargspec(func)  # pylint: disable=deprecated-method
+            spec = inspect.getfullargspec(func)
```

Comment:
We should refactor this to use `inspect.signature()`.  This is a python 3 only api that also returns the appropriate annotations.

See the function and definition of the Signature class here:
https://docs.python.org/3/library/inspect.html#inspect.signature

We support a minimum python version of 3.5+ currently in typedargs so its safe to use `inspect.signature`.

In particular this resolve the `fixme` below about having trouble with`__init__` annotations since the `signature` routine takes in any callable and finds the appropriate information.

Reply:
@timburke 
Refactored in this commit:
https://github.com/iotile/typedargs/pull/69/commits/e797b45c4ae9e5af047d3d18d615c37d5a539f0f